### PR TITLE
Hold note timer

### PR DESCRIPTION
**Pull request by Mello-Cello**
It might be helpful to look at the last two images here: https://tinyurl.com/yxgkoc89

### CODE REVIEW

File: src/modules/helperFunctions.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export function devLogger(...args) {
+    if (process.env.NODE_ENV !== 'development') {
+        return;
+    } else {
```
Comment (DanielOrmeno) [inline src/modules/helperFunctions.js]: this `else` is redundant. you can remove it to keep the code simpler.

File: src/App.js [hunk 2]
```diff
@@ -16,11 +17,12 @@ class App extends Component {
 
       notesPlayedCorrectly: 0,
       notesSkipped: 0,
+      notesTried: -1,
       timeStarted: null,
       timeStopped: null,
       timePlayed: '',
     }
-    console.log(process.env.NODE_ENV);
```
Comment (DanielOrmeno) [inline src/App.js]: should we delete this?

File: src/components/Game.js [hunk 3]
```diff
@@ -15,108 +16,194 @@ class Game extends Component {
...
-      this.setState({userPlayingPitch: pitch}, ()=>{
-        // console.log('in game, userPlayingPitch: ', this.state.userPlayingPitch);
-        if (pitch <= this.state.targetFreqRangeUpper && pitch >= this.state.targetFreqRangeLower) {
-          this.toggleNoteMatched();
+  handlePitchChange = (pitch) => {
+    let pitchInRange = this.checkPitchInRange(pitch); // true or false
+    let timerStarted = this.checkTimerStarted();
+    let targetTimeReached = this.checkTargetTimeReached();
+
+    if (pitchInRange) { // A
```
Comment (DanielOrmeno) [inline src/components/Game.js]: this logic is a bit involved, can we simplify it? its hard to read hence hard to maintain.

File: src/components/Game.js [hunk 4]
```diff
@@ -15,108 +16,194 @@ class Game extends Component {
...
 
-    // add an else here to reset state to false if the time requirement isn't reached
+  };
+
+  handlePitchNoChange(pitch) { // C
+    if (this.state.targetTime && this.checkTargetTimeReached()) { // C1
+      this.generateSparkles();
+      devLogger('sparkles in C1');
+    } else { // C2
+      if (pitch) {
```
Comment (DanielOrmeno) [inline src/components/Game.js]: flatten these confitionals `else if (pitch)`

File: src/components/Game.js [hunk 5]
```diff
@@ -15,108 +16,194 @@ class Game extends Component {
...
-  toggleNoteMatched = () => {
-    // this.state.noteMatched ? this.setState({noteMatched: false}) : this.setState({noteMatched: true})
-    this.setState({noteMatched: true},()=>{
-      console.log(this.state.noteMatched);
-    });
+  checkTargetTimeReached = () => {
+    const currentTime = new Date().getTime();
+    const targetTime = this.state.targetTime;
+
+    if (currentTime >= targetTime) {return true} else {return false};
```
Comment (DanielOrmeno) [inline src/components/Game.js]: `return currentTime >= targetTime`

File: src/components/Start.js [hunk 6]
```diff
@@ -2,6 +2,8 @@ import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import Info from './Info';
 
+// import {devLogger} from '../modules/helperFunctions.js';
```
Comment (DanielOrmeno) [inline src/components/Start.js]: delete?

File: src/components/Summary.js [hunk 7]
```diff
@@ -1,12 +1,14 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+// import {devLogger} from '../modules/helperFunctions.js';
```
Comment (DanielOrmeno) [inline src/components/Summary.js]: delete?

File: src/components/Game.js (hunk 3 above)
Comment (Mello-Cello) [reply src/components/Game.js]: Let's put this on our list of things to discuss on Friday. The whole decision tree is pretty complex.

File: src/components/Game.js (hunk 5 above)
Comment (Mello-Cello) [reply src/components/Game.js]: Ahhh thank you! I knew there was a way to simplify this and couldn't remember in the moment.

File: src/App.js (hunk 2 above)
Comment (Mello-Cello) [reply src/App.js]: sure (done)

File: src/components/Start.js (hunk 6 above)
Comment (Mello-Cello) [reply src/components/Start.js]: sure (done)

File: src/components/Summary.js (hunk 7 above)
Comment (Mello-Cello) [reply src/components/Summary.js]: No, I need this for my next wave.

File: src/components/Game.js (hunk 3 above)
Comment (DanielOrmeno) [reply src/components/Game.js]: Sure, hoever keen in mind that complex logic should not mean complex code ;)

**Pull request merged**